Sources/Public/CalendarView.swift

```diff
@@ -1062,13 +1076,26 @@ extension CalendarView: WidthDependentIntrinsicContentHeightProviding {
   // This is where we perform our width-dependent height calculation. See `DoubleLayoutPassHelpers`
   // for more details about why this is needed and how it works.
   func intrinsicContentSize(forHorizontallyInsetWidth width: CGFloat) -> CGSize {
-    // Force layout so that `visibleItemsDetails.maxMonthHeight` is calculated.
-    setNeedsLayout()
-    layoutIfNeeded()
+    let calendarWidth = width + layoutMargins.left + layoutMargins.right
+    let visibleItemsProvider = VisibleItemsProvider(
+      calendar: calendar,
+      content: content,
+      size: CGSize(width: calendarWidth, height: .maxLayoutValue),
+      layoutMargins: directionalLayoutMargins,
+      scale: scale,
+      monthHeaderHeight: monthHeaderHeight(calendarWidth: calendarWidth),
+      backgroundColor: backgroundColor)
+
+    let anchorMonthHeaderLayoutItem = anchorLayoutItem(
+      for: .init(
+        targetItem: .month(content.monthRange.lowerBound),
+        scrollPosition: .firstFullyVisiblePosition,
+        animated: false),
+      visibleItemsProvider: visibleItemsProvider)
 
-    guard let visibleItemsDetails = visibleItemsDetails else {
-      return CGSize(width: UIView.noIntrinsicMetric, height: UIView.noIntrinsicMetric)
-    }
+    let visibleItemsDetails = visibleItemsProvider.detailsForVisibleItems(
+      surroundingPreviouslyVisibleLayoutItem: anchorMonthHeaderLayoutItem,
+      offset: scrollView.contentOffset)
```

Comment:
previously, we used to force layout here. This would cause layoutSubviews to run. which would use the `visibleItemsProvider` to calculate and return layout info (including a `maxMonthHeight` property, read below).

Forcing layout here when sizing a view in SwiftUI caused crashes in SwiftUI code. Rather than depending on `layoutSubviews` getting called to compute the `intrinsicContentSize`, we can instead manually invoke `visibleItemsProvider.detailsForVisibleItems(...)` to get `visibleItemsDetails.maxMonthHeight`.